### PR TITLE
[14.0.x] ISPN-14902 Exception is thrown while rebuilding an index

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/ISPN-14902
https://issues.redhat.com/browse/JDG-6256

Backport of https://github.com/infinispan/infinispan/pull/10992

### CODE REVIEW

**Comment by wburns** [comment]
Looks like at least 2 related test failures :( https://ci.infinispan.org/job/Infinispan/job/PR-10991/4/

**Comment by fax4ever** [comment]
> Looks like at least 2 related test failures :( https://ci.infinispan.org/job/Infinispan/job/PR-10991/4/

I converted it to draft. I think that we need much time to solve the issue properly.

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LazySearchMapping.java [hunk 1]
```diff
@@ -116,13 +117,37 @@ public Collection<? extends SearchIndexedEntity> allIndexedEntities() {
...
+   }
+
    @Override
    public Set<String> allIndexedEntityNames() {
       return mapping().allIndexedEntityNames();
    }
 
    @Override
    public Set<Class<?>> allIndexedEntityJavaClasses() {
+      if (!searchMappingRef.available()) {
```
Comment (wburns) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LazySearchMapping.java:144]: Looking at this class I an not sure what the stamped lock is protecting. It seems we can do the same `searchMappingRef.available()` call in indexedEntitiesForStatistics then?

**Review by wburns — COMMENTED** [review]
Some minor things, more curious

File: query/src/main/java/org/infinispan/query/stats/impl/LocalIndexStatistics.java [hunk 2]
```diff
@@ -62,15 +69,48 @@ public CompletionStage<Map<String, IndexInfo>> computeIndexInfos() {
...
 
       return countStage
-            .thenCompose(count -> indexedEntity.indexManager().unwrap(LuceneIndexManager.class).computeSizeInBytesAsync()
-            .thenApply(size -> new IndexInfo(count, size)));
+            .thenCompose(count -> {
+               CompletionStage<Long> sizeStage;
+               if (count == -1L) {
+                  sizeStage = CompletableFuture.completedFuture(-1L);
+               } else if (reindexing()) {
+                  sizeStage = CompletableFuture.completedFuture(-1L);
```
Comment (wburns) [inline query/src/main/java/org/infinispan/query/stats/impl/LocalIndexStatistics.java:91]:
```suggestion
               if (count == -1L || reindexing()) {
                  sizeStage = CompletableFuture.completedFuture(-1L);
```

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LazySearchMapping.java (hunk 1 above)
Comment (fax4ever) [reply remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LazySearchMapping.java:144]: You right! Going to do that

**Comment by fax4ever** [comment]
Ci looks good.

**Review by wburns — APPROVED** [review]
LGTM, waiting for 14.0.x to be safe to merge to

**Comment by ryanemerson** [comment]
Thanks @wburns

**Comment by fax4ever** [comment]
Thanks @ryanemerson for merging it

**Pull request merged**